### DISCUSSION TITLE
GitHub PRs blocked by "Review required" when using merge queues even with countermeasures

### DISCUSSION THREAD

**Discussion by travisshivers** [renovatebot/renovate#28156] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub, Renovate 37.272.0

### Please tell us more about your question or problem

Renovate's PRs are blocked by "Review required" and are not able to automerge on GitHub when a merge queue is enabled and "Require approvals" is enabled.

This is probably a GitHub issue or limitation, but am curious if it can be worked around in other ways or configured differently.

The setup I tested for branch protection is:
- Require a pull request before merging
- Require approvals: 1
- Allow specified actors to bypass required pull requests: the app
- Require status checks to pass before merging

I have also tried the same thing with rulesets and it still hits the same issue:
1. Main ruleset with “Require a pull request before merging", "Require merge queue", “Required approvals” set to 0, and "Require status checks to pass"
2. Ruleset with “Require a pull request before merging” and “Required approvals” set to 1, but with the app specified in the bypass list so it shouldn’t be affected by this ruleset.

In both cases, the result is that renovate properly creates the PRs and enables automerge, but the PRs appear blocked in the UI with "Review required" even though it seems like they shouldn't.

**Comment by rarkins** [comment]
You can self host https://github.com/renovatebot/renovate-approve-bot

**Comment by usmonster** [comment]
> I have also tried the same thing with rulesets and it still hits the same issue:
> [...] 
> In both cases, the result is that renovate properly creates the PRs and enables automerge, but the PRs appear blocked in the UI with "Review required" even though it seems like they shouldn't.

We're running into this as well (also similar to #25065), and it's smelling more like a bug or design limitation in GitHub's merge queue feature than a Renovate-specific issue.

Essentially, it appears that no user/app can add a PR to the queue until it's "passed all required branch protection checks", even if said user/app has been explicitly allowed to bypass said checks. If this is indeed the case, then I'm not sure there's a generic workaround for this, so I'll raise an issue with GitHub Support and comment on the issue in [their feedback forum](https://github.com/orgs/community/discussions/categories/pull-requests?discussions_q=category%3A%22Pull+Requests%22+%22merge+queue%22).

**Reply by usmonster** [reply]
> [!NOTE]
> Raised the issue in [a GitHub discussion here](https://github.com/orgs/community/discussions/110908#discussioncomment-9688029).

**Comment by dge808** [comment]
hey @travisshivers were you able to find a workaround for this? I am experiencing exactly the same issue.

**Reply by ocofaigh** [reply]
I also have hit this. As far as I can see, the only options are to:
A) Use the inbuilt renovate auto merge feature and add the renovate actor to the bypass list. It means you have to wait for renovate to run again to until the PRs are merged though
B) Self host https://github.com/renovatebot/renovate-approve-bot if you are using GHE (or use the Mend hosted version for public github)
